app/models/mixins/assignment_mixin.rb

```diff
@@ -81,17 +81,25 @@ def remove_all_assigned_tos(cat = nil)
   delegate :namespace, :to => :class
 
   module ClassMethods
+    # get a mapping of alert_sets and the tags they are assigned
+    #
+    # the namespace is removed from the front of the alert_set's tag
+    #   alert_set will have:
+    #     alert_set.tag.name == "/miq_alert_set/assigned_to/vm/tag/managed/environment/test"
+    #   will return
+    #     {assigned_to: "vm/tag/managed/environment/test"}
+    #   and will match:
+    #     vm.tag.name = "/managed/environment/test"
     def assignments
       # Get all assigned, enabled instances for type klass
       records = kind_of?(Class) ? all : self
       assignment_map = records.each_with_object({}) { |a, h| h[a.id] = a }
       Tag
         .where("taggings.taggable_type = ? and tags.name like ?", name, "#{namespace}/%")
-        .joins(:taggings)
-        .each_with_object([]) do |tag, arr|
-          tag.taggings.each do |tagging|
-            next unless assignment_map[tagging.taggable_id]
-            arr << {
+        .includes(:taggings).references(:taggings)
```

Comment:
Can you either put the `.references` on it's own line, or (even more preferable for me) above the `.where` clause?  I find it weird, personally, to see the references call *after* the bit that needs it, but perhaps that's just me.